core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -53,7 +53,8 @@ namespace Kokkos {
 namespace Experimental {
 namespace Impl {
 
-int SYCLInternal::was_finalized = 0;
+std::vector<std::optional<sycl::queue>*> SYCLInternal::all_queues;
+std::mutex SYCLInternal::mutex;
 
 SYCLInternal::~SYCLInternal() {
```

Comment:
The destructor should abort if `was_finalized` isn't true (or call `finalize()`).  Otherwise, we have a dangling `m_queue` inside `all_queues`.

Reply:
We already print to `std::cerr` which is consistent with the other backends. I added a check to `was_finalized`, though.